app/services/orchestration/task.rb

```diff
@@ -39,7 +45,14 @@ def update_ts
 
     # sort based on priority
     def <=>(other)
+      return self.created <=> other.created if self.priority == other.priority
       self.priority <=> other.priority
     end
+
+    def ==(other)
+      return false unless other.is_a?(Task)
+      return self.id == other.id if self.id && other.id
+      self.name == other.name
```

Comment:
if `id` is now required, why we fallback to name?

Reply:
now I get it after reading your last comment, can't we then use name in initializer as an id if id was not passed as argument? therefore we wouldn't need fallback here (and potentially other places)

Reply:
Absolutely good idea that's also faster!